install.sh

```diff
@@ -137,33 +142,39 @@ fi
 
 # Lets register the agent
 if [ ! -z "${MONDOO_REGISTRATION_TOKEN}" ]; then
-
-  if [ $(cat /proc/1/comm) = "init" ]
-  then
-    echo " -> Stop mondoo upstart service"
-    $sudo_cmd stop mondoo || true
-  elif [ $(cat /proc/1/comm) = "systemd" ]
-  then
-    echo " -> Stop mondoo systemd service"
-    $sudo_cmd systemctl stop mondoo
-  fi
-
-  purple_bold "\n* Register agent with Mondoo Cloud"
-  $sudo_cmd mkdir -p /etc/opt/mondoo/
-  $sudo_cmd mondoo register --config /etc/opt/mondoo/mondoo.yml --token $MONDOO_REGISTRATION_TOKEN
-
-  if [ $(cat /proc/1/comm) = "init" ]
-  then
-    purple_bold "\n* Configuring upstart service"
-    $sudo_cmd start mondoo || true
-  elif [ $(cat /proc/1/comm) = "systemd" ]
-  then
-    purple_bold "\n* Configuring systemd service"
-    $sudo_cmd systemctl enable mondoo.service
-    $sudo_cmd systemctl start mondoo.service
-    $sudo_cmd systemctl daemon-reload
+  # check the exit code of mondoo status command
+  if [ $(mondoo status >nul 2>&1; echo $?) -eq "0" ]; then
```

Comment:
this one threw me for a bit. because we have an error trap up above we can't just run the command in a separate line and then eval the exit code, because the 1 exit code gets recognized as an error and the script exits, so we have to do it all in one line

Reply:
Good finding. There potentially another case where the network connection is not working. Lets cover this later since its unlikely and the worst case is that the agent gets new credentials